docs/developer-tools/Guard.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Guard
+author: Sergio0694
+description: Helper methods to verify conditions when running code
+keywords: windows 10, uwp, windows community toolkit, uwp community toolkit, uwp toolkit, debug, net core, net standard
+dev_langs:
+  - csharp
+---
+
+# Guard
+
+The [Guard](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.diagnostics.guard) can be used to validate method arguments in a streamlined manner, which is also faster, less verbose, more expressive and less error prone than manually writing checks and throwing exceptions.
+
+## How it works
+
+These `Guard` APIs are built with three core principles in mind:
+
+- Being **fast**. To achieve that, all the APIs have been designed to produce as little code as possible in the caller, and each single Guard API will (almost always) be inlined. Furthermore, specialized methods are generated with T4 templates to achieve the most efficient assembly code possible when working with common types (eg. primitive numeric types).
+- Being **helpful**. Each `Guard` API produces a detailed exception message that clearly specifies what went wrong, along with additional info (eg. current variable values), when applicable.
+- Being **intuitive**. To achieve this, all the `Guard` APIs have expressive and self-explanatory names that make it immediately clear what each API is supposed to do. This shifts the burden of writing checks away from the developers, letting them express conditions using natural language.
+
+## Syntax
+
+Here is a sample method, with some checks being done with explicit checks and manual throw statements:
+
+```csharp
```

Comment:
🦙❤ This example juxtaposition is amazing! Thanks for going to this level of detail!

Reply:
No worries, glad you liked how I structured it!
I thought it could be useful to introduce new people to the APIs and make it immediately clear how they can be used and most importantly why they should use them 😄